### PR TITLE
Mobile tests for Sauce Labs App

### PR DESCRIPTION
**Pull request by asurelaba**

### CODE REVIEW

File: src/main/java/com/solvd/carina/mobile/android/CartScreen.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        removeButton = removeButton.format(product);
+        swipe(removeButton, 3);
+        removeButton.click();
+    }
+
+    @Override
+    public CheckoutInformationScreenBase clickCheckoutButton() {
+        if (swipe(checkoutButton, 5)) {
+            checkoutButton.click();
+        }
```
Comment (ankochergin) [inline src/main/java/com/solvd/carina/mobile/android/CartScreen.java]:
```suggestion
        swipe(checkoutButton, 5)
        checkoutButton.click();
```
it'll stop swiping when element appears on the screen

File: src/main/java/com/solvd/carina/mobile/android/ProductScreen.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.openqa.selenium.support.FindBy;
+
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = ProductScreenBase.class)
+public class ProductScreen extends ProductScreenBase {
+
+    @FindBy(xpath = "//android.widget.TextView[@text='PRODUCTS']")
+    private ExtendedWebElement productScreenTitle;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-Menu\"]/android.view.ViewGroup/android.widget.ImageView")
```
Comment (ankochergin) [inline src/main/java/com/solvd/carina/mobile/android/ProductScreen.java]: does it find element with this locator?
```suggestion
    @FindBy(xpath = "//android.view.ViewGroup[@content-desc=\"test-Menu\"]")
```

Comment (ankochergin) [reply src/main/java/com/solvd/carina/mobile/android/ProductScreen.java]: if yes - please update all locators and make them shorter if possible

File: src/main/java/com/solvd/carina/utils/mobile/NavigationUtils.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+public class NavigationUtils implements ICustomTypePageFactory {
+
+    private ProductScreenBase login() {
+        LoginScreenBase loginScreen = initPage(getDriver(), LoginScreenBase.class);
+        loginScreen.typeUsername(R.TESTDATA.get("mobile.username"));
+        loginScreen.typePassword(R.TESTDATA.get("mobile.password"));
+        loginScreen.isOpened();
+        return loginScreen.clickLoginButton();
+    }
```
Comment (ankochergin) [inline src/main/java/com/solvd/carina/utils/mobile/NavigationUtils.java]: extract login to a separate class, authentication and navigation should be separately. 1 util class should do only 1 type of actions

File: src/main/java/com/solvd/carina/utils/mobile/NavigationUtils.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public CartScreenBase navigateToCartWithItems() {
+        return navigateToProductScreenAfterItemsAddedToCart().clickCartIcon();
+    }
+
+    public ProductScreenBase navigateToProductScreenAfterItemsAddedToCart() {
+        ProductScreenBase productScreen = login();
+        String product = "Sauce Labs Backpack";
+        productScreen.clickAddToCartButton(product);
+        return productScreen;
+    }
```
Comment (ankochergin) [inline src/main/java/com/solvd/carina/utils/mobile/NavigationUtils.java]: I think hardcoding product title in utils is not really good, could you please think about some other solution?

File: src/main/java/com/solvd/carina/mobile/base/MenuScreenBase.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+public abstract class MenuScreenBase extends MobileAbstractPage {
+
+    public MenuScreenBase(WebDriver driver) {
+        super(driver);
+    }
+
+    public abstract LoginScreenBase clickLogoutText();
+
+    public abstract WebViewScreenBase clickAboutText();
```
Comment (ankochergin) [inline src/main/java/com/solvd/carina/mobile/base/MenuScreenBase.java]:
```suggestion
    public abstract LoginScreenBase clickLogoutLink();

    public abstract WebViewScreenBase clickAboutLink();
```
or `Button`

File: src/main/java/com/solvd/carina/mobile/android/CartScreen.java (hunk 1 above)
Comment (asurelaba) [reply src/main/java/com/solvd/carina/mobile/android/CartScreen.java:46]: @ankochergin Hi, if i want to check if the checkout button is present first before clicking, do i need to do 
checkoutButton.isElementPresent();
swipe(checkoutButton,5);
checkoutButton.click();

**Pull request merged**